### PR TITLE
Adds a new, sick way to unmount a skateboar

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Adds a new ui action for the skateboard, for a kickflip. On use, you will jump in place and "catch" the skateboard in midair, putting it in your hands.

## Why It's Good For The Game
Unmounting skateboards is TEDIOUS, the sprite is too small to reliably click drag it. Considering you can quickly mount a skateboard from the held item version, it makes sense that you can have an option to quickly get it in your hands.
Its also slick as hell.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/73374039/184525670-a9630e37-917a-4d32-b874-7a9a3868fd58.mp4

</details>

## Changelog
:cl:
add: Nanotrasen has implemented kickflip skills into all employees, in an attempt to boost skateboard sales.
/:cl:

### CODE REVIEW

**Comment by etherware-novice** [comment]
testing files coming, obs is being very difficult rn

**Comment by Fronsis** [comment]
Damn that looks dope, guess it's missing the sound and maybe an arrow pointing downwards to differentiate from the 1st button?

**Comment by RDS88-beeman** [comment]
My mechanics senses are tingling, what happens if a xeno kickflips a skateboard. Does it go into their hands?

**Comment by etherware-novice** [comment]
> Damn that looks dope, guess it's missing the sound and maybe an arrow pointing downwards to differentiate from the 1st button?

i dobt know how to edit the sprite to look belivable so no

**Comment by etherware-novice** [comment]
> My mechanics senses are tingling, what happens if a xeno kickflips a skateboard. Does it go into their hands?

it calls the skateboards local pickup proc so i think they cant

**Comment by itsmeow** [comment]
Add the kickflip sound please

File: code/modules/vehicles/vehicle_actions.dm [hunk 1]
```diff
@@ -228,3 +228,20 @@
...
+/datum/action/vehicle/ridden/scooter/skateboard/kflip
+	name = "Kick Flip"
+	desc = "Do a sweet kickflip to dismount..in style."
+	button_icon_state = "skateboard_ollie"
+
+/datum/action/vehicle/ridden/scooter/skateboard/kflip/Trigger()
+	var/obj/vehicle/ridden/scooter/skateboard/V = vehicle_target
+	var/mob/living/L = owner
+
+	L.visible_message("[L] does a neat kickflip and catches [L.p_their()] board in midair.", "<span class='notice'>You do a sick kickflip, catching the board in midair! Stylish.")
```
Comment (ivanmixo) [inline code/modules/vehicles/vehicle_actions.dm]:
```suggestion
	L.visible_message("[L] does a neat kickflip and catches [L.p_their()] board in midair.", "<span class='notice'>You do a sick kickflip, catching the board in midair! Stylish.</span>")
```

File: code/modules/vehicles/vehicle_actions.dm [hunk 2]
```diff
@@ -228,3 +228,20 @@
 			V.icon_state = "[V.board_icon]-grind"
 			addtimer(CALLBACK(V, /obj/vehicle/ridden/scooter/skateboard/.proc/grind), 2)
 		next_ollie = world.time + 5
+
+/datum/action/vehicle/ridden/scooter/skateboard/kflip
+	name = "Kick Flip"
+	desc = "Do a sweet kickflip to dismount..in style."
```
Comment (ivanmixo) [inline code/modules/vehicles/vehicle_actions.dm]:
```suggestion
	desc = "Do a sweet kickflip to dismount... in style."
```
Nitpick but please.

**Review by ivanmixo — CHANGES_REQUESTED** [review]

**Comment by itsmeow** [comment]
Does this have a failure chance like the ollie? It probably should

**Comment by etherware-novice** [comment]
the ollie always succeeds unless you hit max stamina damage, so maybe ill just make it use 5 stam to kickflip?

**Comment by itsmeow** [comment]
yeah it should use stam like ollies

**Comment by itsmeow** [comment]
Don't add brute damage, please...

File: code/modules/vehicles/vehicle_actions.dm [hunk 3]
```diff
@@ -228,3 +228,28 @@
...
+
+	L.adjustStaminaLoss(V.instability)
+	if (L.getStaminaLoss() >= 100)
+		playsound(src, 'sound/effects/bang.ogg', 20, TRUE)
+		V.unbuckle_mob(L)
+		L.Paralyze(50)
+		L.adjustBruteLoss(10)  // thats gonna leave a mark
+		V.visible_message("<span class='userdanger'>You fall flat onto the board!</span>","<span class='danger'>[L] misses the landing and falls on [L.p_their()] face!</span>")
+	else
+		L.visible_message("[L] does a neat kickflip and catches [L.p_their()] board in midair.", "<span class='notice'>You do a sick kickflip, catching the board in midair! Stylish.")
```
Comment (itsmeow) [inline code/modules/vehicles/vehicle_actions.dm]: The visible message should say sick or sweet, not neat

**Comment by etherware-novice** [comment]
how about a 5-10 percent chance to cause brute

File: code/modules/vehicles/vehicle_actions.dm [hunk 4]
```diff
@@ -228,3 +228,28 @@
...
+/datum/action/vehicle/ridden/scooter/skateboard/kflip/Trigger()
+	var/obj/vehicle/ridden/scooter/skateboard/V = vehicle_target
+	var/mob/living/L = owner
+
+	L.adjustStaminaLoss(V.instability)
+	if (L.getStaminaLoss() >= 100)
+		playsound(src, 'sound/effects/bang.ogg', 20, TRUE)
+		V.unbuckle_mob(L)
+		L.Paralyze(50)
+		L.adjustBruteLoss(10)  // thats gonna leave a mark
```
Comment (itsmeow) [inline code/modules/vehicles/vehicle_actions.dm]: no

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/vehicle_actions.dm (hunk 4 above)
Comment (etherware-novice) [reply code/modules/vehicles/vehicle_actions.dm]: 5-10% chance?

Comment (ivanmixo) [reply code/modules/vehicles/vehicle_actions.dm]: This is good, wdym lol

File: code/modules/vehicles/vehicle_actions.dm (hunk 3 above)
Comment (PowerfulBacon) [inline code/modules/vehicles/vehicle_actions.dm]: Close your spans and add a span for the first one.

**Review by itsmeow — APPROVED** [review]

**Review by ike709 — APPROVED** [review]

**Review by Archanial — APPROVED** [review]

**Comment by itsmeow** [comment]
# The skateboar is real

image

**Pull request merged**